src/write_concern.ts

```diff
@@ -35,41 +56,66 @@ export const WRITE_CONCERN_KEYS = ['w', 'wtimeout', 'j', 'journal', 'fsync'];
  * @see https://www.mongodb.com/docs/manual/reference/write-concern/
  */
 export class WriteConcern {
-  /** request acknowledgment that the write operation has propagated to a specified number of mongod instances or to mongod instances with specified tags. */
-  w?: W;
-  /** specify a time limit to prevent write operations from blocking indefinitely */
+  /** Request acknowledgment that the write operation has propagated to a specified number of mongod instances or to mongod instances with specified tags. */
+  readonly w?: W;
+  /** Request acknowledgment that the write operation has been written to the on-disk journal */
+  readonly journal?: boolean;
+  /** Specify a time limit to prevent write operations from blocking indefinitely */
+  readonly wtimeoutMS?: number;
+  /**
+   * Specify a time limit to prevent write operations from blocking indefinitely.
+   * @deprecated Will be removed in the next major version. Please use wtimeoutMS.
+   */
   wtimeout?: number;
-  /** request acknowledgment that the write operation has been written to the on-disk journal */
+  /**
+   * Request acknowledgment that the write operation has been written to the on-disk journal.
+   * @deprecated Will be removed in the next major version. Please use journal.
+   */
   j?: boolean;
-  /** equivalent to the j option */
+  /**
+   * Equivalent to the j option.
+   * @deprecated Will be removed in the next major version. Please use journal.
+   */
   fsync?: boolean | 1;
 
   /**
    * Constructs a WriteConcern from the write concern properties.
    * @param w - request acknowledgment that the write operation has propagated to a specified number of mongod instances or to mongod instances with specified tags.
-   * @param wtimeout - specify a time limit to prevent write operations from blocking indefinitely
-   * @param j - request acknowledgment that the write operation has been written to the on-disk journal
+   * @param wtimeoutMS - specify a time limit to prevent write operations from blocking indefinitely
+   * @param journal - request acknowledgment that the write operation has been written to the on-disk journal
    * @param fsync - equivalent to the j option
    */
-  constructor(w?: W, wtimeout?: number, j?: boolean, fsync?: boolean | 1) {
+  constructor(w?: W, wtimeoutMS?: number, journal?: boolean, fsync?: boolean | 1) {
     if (w != null) {
       if (!Number.isNaN(Number(w))) {
         this.w = Number(w);
       } else {
         this.w = w;
       }
     }
-    if (wtimeout != null) {
-      this.wtimeout = wtimeout;
+    if (wtimeoutMS != null) {
+      this.wtimeoutMS = this.wtimeout = wtimeoutMS;
     }
-    if (j != null) {
-      this.j = j;
+    if (journal != null) {
+      this.journal = this.j = journal;
     }
     if (fsync != null) {
-      this.fsync = fsync;
+      this.journal = this.j = fsync ? true : false;
     }
   }
 
+  /**
+   * Apply a write concern to a command document.
```

Comment:
```suggestion
   * Apply a write concern to a command document. Will modify and return the command.
```
Just a bit more docs, and double checking this is supposed to be public correct? 

I also notice we don't use the return value, just the modification, should we pick one instead? (also fine with both, similar to `Object.assign` behavior, just asking the Q)